modules/c++/sys/unittests/test_symlink.cpp

```diff
@@ -67,18 +69,18 @@ TEST_CASE(testCreateSymlink)
     catch (const std::exception& ex)
     {
         std::cerr << "Caught std::exception: " << ex.what() << std::endl;
-        TEST_ASSERT(false);

```

Comment:
Adding /wd4127 to the command-line makes it harder to get to the day when the code is free of the "conditional expression is constant" warning, and no new code can be added which generates that warning because "warnings as errors" is enabled.

Of course, it's also perfectly reasonable to say "we'll never, ever care about this diagnostic" and always turn it off.  I don't like doing that because I figure the compiler writer is smarter than I am and the warning is there for a reason.  Maybe not a very big or important reason (right now), but something that I should perhaps think about some day.